### PR TITLE
chore: Importing auto-generated code from googleapis-gen

### PR DESCRIPTION
**Pull request by jainsahab**
These changes consists of:
- Setting up a shell script to simulate Owlbot and download generated code from googleapis-gen.
- Owlbot config which Owlbot uses  to copy generated code to the datastore repo.
- Add/Update maven configuration for new modules

### CODE REVIEW

**Review by kolea2 — CHANGES_REQUESTED** [review]
generally looks great! Just please ensure all version numbers are consistent snapshots

**Review by kolea2 — COMMENTED** [review]
another comment/idea - in a different PR for `main`, please make these repo checks required in https://github.com/googleapis/java-datastore/blob/main/.github/sync-repo-settings.yaml so we don't accidentally merge any breaking changes

File: grpc-google-cloud-datastore-v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.api.grpc</groupId>
+  <artifactId>grpc-google-cloud-datastore-v1</artifactId>
+  <version>2.17.5</version><!-- {x-version-update:grpc-google-cloud-datastore-v1:current} -->
```
[the remark this answers is not in the document]
Comment (jainsahab) [reply grpc-google-cloud-datastore-v1/pom.xml]: Fixed

File: google-cloud-datastore-bom/pom.xml [hunk 2]
```diff
@@ -54,6 +54,11 @@
         <artifactId>google-cloud-datastore</artifactId>
         <version>2.17.6-SNAPSHOT</version><!-- {x-version-update:google-cloud-datastore:current} -->
       </dependency>
+      <dependency>
+        <groupId>com.google.api.grpc</groupId>
+        <artifactId>grpc-google-cloud-datastore-v1</artifactId>
+        <version>2.17.5</version><!-- {x-version-update:grpc-google-cloud-datastore-v1:current} -->
```
[the remark this answers is not in the document]
Comment (jainsahab) [reply google-cloud-datastore-bom/pom.xml]: Fixed

File: grpc-google-cloud-datastore-v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.api.grpc</groupId>
+  <artifactId>grpc-google-cloud-datastore-v1</artifactId>
+  <version>2.17.6-SNAPSHOT</version><!-- {x-version-update:grpc-google-cloud-datastore-v1:current} -->
+  <name>grpc-google-cloud-datastore-v1</name>
+  <description>GRPC library for google-cloud-datastore</description>
+  <parent>
+    <groupId>com.google.cloud</groupId>
+    <artifactId>google-cloud-datastore-parent</artifactId>
+    <version>2.17.5</version><!-- {x-version-update:google-cloud-datastore:current} -->
```
Comment (kolea2) [inline grpc-google-cloud-datastore-v1/pom.xml:13]: one last SNAPSHOT update to latest parent

**Review by kolea2 — APPROVED** [review]
LGTM after last nit - I reviewed the POMs and other non autogen files, but lmk if I should take a closer look at any of the others

File: grpc-google-cloud-datastore-v1/pom.xml (hunk 3 above)
Comment (jainsahab) [reply grpc-google-cloud-datastore-v1/pom.xml]: Fixed

**Review by kolea2 — APPROVED** [review]

**Pull request merged**